### PR TITLE
Phase2 L1T, Correlator Layer-2 : support emulation of with TMUX18 input + GT interface tweaks

### PR DESCRIPTION
**Pull request by cerminar**
#### PR description:

The PR introduces the following changes:
- support pattern file creation for Correlator Layer-2 e/gamma boards with TMUX18 Layer-1 inputs.
- tweak the GT interface, renaming the `isolation` and `quality` methods of HW GT object according to the new convention.
- set the charge of non-valid HW GT electron objects to 0.
- address precision of the `z0` parameter in TkElectron to have the same precision as in HW

#### PR validation:

No expected change in performance: changes are only related to interfaces with emulator code.

This is equivalent to: 
https://github.com/cms-l1t-offline/cmssw/pull/1215

### CODE REVIEW

**Comment by epalencia** [comment]
Please test

**Comment by epalencia** [comment]
Please test

File: DataFormats/L1TParticleFlow/interface/egamma.h [hunk 1]
```diff
@@ -166,11 +166,11 @@ namespace l1ct {
       ele.v3.pt = CTtoGT_pt(hwPt);
       ele.v3.phi = CTtoGT_phi(hwVtxPhi());
       ele.v3.eta = CTtoGT_eta(hwVtxEta());
-      ele.quality = hwQual;
+      ele.qualityFlags = hwQual;
       // NOTE: GT:  0 = positive, 1 = negative, CT: 0 = negative, 1 = positive
-      ele.charge = !hwCharge;
+      ele.charge = !hwCharge & ele.valid;
```
Comment (aloeliger) [inline DataFormats/L1TParticleFlow/interface/egamma.h:171]:
```suggestion
      ele.charge = (!hwCharge) & ele.valid;
```

**Comment by aloeliger** [comment]
+l1

**Comment by epalencia** [comment]
ping @cms-sw/upgrade-l2

**Comment by srimanob** [comment]
+Upgrade

**Comment by antoniovilela** [comment]
+1

**Pull request merged**